src/core.js

```diff
@@ -2059,26 +2059,8 @@ define([
     };
 
     function getWidgetClassAndOptions(propPath, mug) {
-        var propDef = mug.p.getDefinition(propPath),
-            propVal = mug.p[propPath];
-
-        // handle properties whose visibility depends on other properties'
-        // visibility (implied visible_if_present on the depending property
-        // takes precedence)
-        if (_.isUndefined(propVal) &&
-            propDef && propDef.visibility &&
-            mug.p.getDefinition(propDef.visibility) &&
-            !getWidgetClassAndOptions(propDef.visibility, mug))
-        {
-            return null;
-        }
-
-        if (!propDef || propDef.visibility === 'hidden' ||
-            (_.isFunction(propDef.visibility) && !propDef.visibility(mug, propDef)) ||
-            (_.isUndefined(propVal) &&
-             (propDef.visibility === "visible_if_present" ||
-              propDef.presence === "notallowed")))
```

Comment:
Note both of the above were added in huge commits where lots of things were happening, so I'm inclined to believe that they were mistakes.